lib/demos/material/dialog_demo.dart

```diff
@@ -35,101 +90,148 @@ class DialogDemo extends StatelessWidget {
     return '';
   }
 
-  Future<void> _showDemoDialog<T>({BuildContext context, Widget child}) async {
-    child = ApplyTextOptions(
-      child: Theme(
-        data: Theme.of(context),
-        child: child,
-      ),
-    );
-    final value = await showDialog<T>(
-      context: context,
-      builder: (context) => child,
-    );
-    // The value passed to Navigator.pop() or null.
-    if (value != null && value is String) {
-      ScaffoldMessenger.of(context).hideCurrentSnackBar();
-      ScaffoldMessenger.of(context).showSnackBar(SnackBar(
-        content:
-            Text(GalleryLocalizations.of(context).dialogSelectedOption(value)),
-      ));
-    }
-  }
-
-  void _showAlertDialog(BuildContext context) {
+  static Route<String> _alertDialogDemoRoute(
+    BuildContext context,
+    Object arguments,
+  ) {
     final theme = Theme.of(context);
     final dialogTextStyle = theme.textTheme.subtitle1
         .copyWith(color: theme.textTheme.caption.color);
-    _showDemoDialog<String>(
+
+    final themes = InheritedTheme.capture(
+      from: context,
+      to: Navigator.of(
+        context,
+        rootNavigator: true,
+      ).context,
+    );
```

Comment:
Why do we need to capture these themes? My understanding is that is is capturing the themes between the Navigator that will own the route and the root Navigator. However, everything above the Navigator that will own the route should already be visible to the route, so there shouldn't be a need to capture?

Reply:
Same question for the instances below.

Reply:
I was just copying the default behavior without attempting to modify anything, but you're right, it's unnecessary. Removed it!